lib/git.zsh

```diff
@@ -127,62 +138,50 @@ function git_prompt_remote() {
 # Formats prompt string for current git commit short SHA
 function git_prompt_short_sha() {
   local SHA
-  SHA=$(command git rev-parse --short HEAD 2> /dev/null) && echo "$ZSH_THEME_GIT_PROMPT_SHA_BEFORE$SHA$ZSH_THEME_GIT_PROMPT_SHA_AFTER"
+  SHA=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) && echo "$ZSH_THEME_GIT_PROMPT_SHA_BEFORE$SHA$ZSH_THEME_GIT_PROMPT_SHA_AFTER"
 }
 
 # Formats prompt string for current git commit long SHA
 function git_prompt_long_sha() {
   local SHA
-  SHA=$(command git rev-parse HEAD 2> /dev/null) && echo "$ZSH_THEME_GIT_PROMPT_SHA_BEFORE$SHA$ZSH_THEME_GIT_PROMPT_SHA_AFTER"
+  SHA=$(__git_prompt_git rev-parse HEAD 2> /dev/null) && echo "$ZSH_THEME_GIT_PROMPT_SHA_BEFORE$SHA$ZSH_THEME_GIT_PROMPT_SHA_AFTER"
 }
 
 # Get the status of the working tree
 function git_prompt_status() {
+  emulate -L zsh
+
   local INDEX STATUS
-  INDEX=$(command git status --porcelain -b 2> /dev/null)
+  INDEX=$(__git_prompt_git status --porcelain -b 2> /dev/null) || return 0
   STATUS=""
-  if $(echo "$INDEX" | command grep -E '^\?\? ' &> /dev/null); then
+  if [[ "${INDEX}" =~ $'(^|\n)\\?\\? ' ]]; then
```

Comment:
This is wrong, it should be `$'(^|\n)?? '`.

Reply:
No, it's not. This is a regex. You can verify with an example status line:
```
INDEX='?? foo'; [[ "${INDEX}" =~ $'(^|\n)\\?\\? ' ]];! echo $?
```
> `0`

Versus the suggestion:
```
INDEX='?? foo'; [[ "${INDEX}" =~ $'(^|\n)?? ' ]];! echo $?
```
> `zsh: failed to compile regex: repetition-operator operand invalid`

I'm very sure that these changes are correct. I've been running with them for a long time.